### PR TITLE
Fix #1728, -Ddubbo.service.delay not work

### LINKED ISSUE #1728: System property dubbo.service.delay invalid
根据dubbo参数的映射关系<dubbo:service delay="5000">也可以通过-Ddubbo.service.delay=5000配置，但是发现-Ddubbo.service.delay=5000不起作用；

debug下面这段代码（ServiceConfig.java）发现：

```java
public synchronized void export() {
    ... ...
    if (delay != null && delay > 0) {
        delayExportExecutor.schedule(new Runnable() {
            @Override
            public void run() {
                doExport();
            }
        }, delay, TimeUnit.MILLISECONDS);
    } else {
        doExport();
    }
}
```

如果有delay，先delay再doExport()，而解析JVM参数的逻辑（appendProperties(this);）是在doExport()方法中调用；所以导致 if (delay != null && delay > 0)判断时，还没有解析JVM参数-Ddubbo.service.delay=5000，从而导致这种配置不起作用；

### PR DESCRIPTION
**Pull request by mizhoux**
## What is the purpose of the change

Fixed issue #1728.

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-dubbo/tree/master/dubbo-test).
- [x] Run `mvn clean install -DskipTests` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java [hunk 1]
```diff
@@ -217,6 +219,33 @@ public void run() {
...
+    /**
+     * Tries to get the value(ms) of the global service delay from the VM options or the properties file.
+     *
+     * @return the value(ms) of the global service delay, <code>null</code> if it not be assigned.
+     */
+    private Integer getGlobalServiceDelay() {
+        // try to get DUBBO_SERVICE_DELAY from the VM options first
+        String delayProperty = System.getProperty(Constants.DUBBO_SERVICE_DELAY_KEY);
+
+        if (delayProperty == null) {
```
Comment (carryxyh) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java]: Maybe use StringUtils is better.
Empty string (like "") or null seems like property delay didn't config.

File: dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java [hunk 2]
```diff
@@ -205,6 +205,8 @@ public synchronized void export() {
             return;
         }
 
+        delay = getGlobalServiceDelay();
```
Comment (carryxyh) [inline dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java]: Your pr may have a problem.
If you directly override the delay here, the delay attribute in the XML file will never take effect.

**Comment by carryxyh** [comment]
Also pls to fix the unit test.

```
Your pr may have a problem.
If you directly override the delay here, the delay attribute in the XML file will never take effect.
```

That is the reason that why is the ut failed.
:)

**Comment by diecui1202** [comment]
I think we need to refactor this to fix the bug, instead of adding `getGlobalServiceDelay`. It's not the best way.

**Pull request closed without merging**